green-square-sequence/src/App.js

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react'
+import 'style.css'
+
+//set how many green squares you want displayed here
+let numOfSquares = 6
+
+export default function App() {
+  let [order, setOrder] = useState([])
+  let squares = [...Array(numOfSquares)]
+
+  const addElement = (num) => {
+    setOrder((order = order.concat(num)))
+  }
+
+  const reverseSequence = async () => {
+    for (let i = 1; i <= numOfSquares; i++) {
+      await delay(500)
+      const newOrder = order.slice(0, order.length - i)
+      setOrder(newOrder)
+    }
+  }
+
+  function delay(time) {
+    return new Promise((res) => setTimeout(res, time))
+  }
+
+  if (order.length >= numOfSquares) {
+    reverseSequence()
```

Comment:
This logic would be better in your onClick event handler.  You have to consider that React components might get randomly re-rendered due to other parts of the App -- so if your component gets re-rendered before the first `delay()` finishes, you'll end up with multiple reverseSequence() loops running at the same time.